src/dominatingset/degree_dom_set.jl

```diff
@@ -0,0 +1,68 @@
+export DegreeDominatingSet
+
+struct DegreeDominatingSet end
+
+"""
+    update_dominated(degree_queue, v, dominated, in_dom_set)
+
+Check if a vertex is already dominated.
+If not, make it dominated and update `degree_queue` by decreasing
+the priority of the vertices adjacent to `v` by 1.
+"""
+function update_dominated(
+    g::AbstractGraph{T},
+    degree_queue::PriorityQueue,
+    v::Integer,
+    dominated::BitArray{1},
+    in_dom_set::BitArray{1}
+    ) where T <: Integer
+
+    @inbounds if !dominated[v]
+        dominated[v] = true
+        if !in_dom_set[v] 
+            degree_queue[v] -= 1
+        end
+        @inbounds @simd for u in neighbors(g, v)
+            if !in_dom_set[u] 
+                degree_queue[u] -= 1
+            end
```

Comment:
I'm surprised, that you can use an if-statement inside a `@simd` loop, I thought the compiler would complain. Still, consider the branchless version
```
degree_queue[u] -= ifelse(in_dom_set[u], 0, 1)
```
This works with self loops, right?

Reply:
I was thinking the same but `@simd` was used in similar situations for `gdistances!`.

Reply:
Also the code you gave won't work since the key `u` may not exist in the queue

Reply:
Yeah, you are right, i though it was an array.